packages/graphql/src/schema-model/library-directives.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { Annotations } from "./annotation/Annotation";
+import type { DEPRECATED } from "../constants";
+import { SHAREABLE } from "../constants";
+
+const additionalDirectives = ["alias", "relationship", "relationshipProperties", "node", SHAREABLE] as const;
+
+export type LibraryDirectives = keyof Annotations | (typeof additionalDirectives)[number] | typeof DEPRECATED;
```

Comment:
I see that `additionalDirectives` is not really used other than for typings, so maybe this whole bit can be simplified?

```suggestion
type AdditionalDirectives = "alias" | "relationship" | "relationshipProperties" | "node" | typeof SHAREABLE | typeof DEPRECATED 

export type LibraryDirectives = keyof Annotations | AdditionalDirectives;
```

Reply:
For the PR #4509 i will need this array, so I want to keep it as is.